lib/propolis-client/src/lib.rs

```diff
@@ -19,106 +38,21 @@ progenitor::generate_api!(
     interface = Builder,
     tags = Separate,
     replace = {
-        PciPath = crate::PciPath,
+        PciPath = crate::instance_spec::PciPath,
+        ReplacementComponent = crate::instance_spec::ReplacementComponent,
+        InstanceSpecV0 = crate::instance_spec::InstanceSpecV0,
+        VersionedInstanceSpec = crate::instance_spec::VersionedInstanceSpec,
     },
     // Automatically derive JsonSchema for instance spec-related types so that
     // they can be reused in sled-agent's API. This can't be done with a
     // `derives = [schemars::JsonSchema]` directive because the `SpecKey` type
     // needs to implement that trait manually (see below).
     patch = {
-        BlobStorageBackend = { derives = [ schemars::JsonSchema ]},
-        Board = { derives = [ schemars::JsonSchema ]},
-        BootOrderEntry = { derives = [ schemars::JsonSchema ]},
-        BootSettings = { derives = [ schemars::JsonSchema, Default] },
-        ComponentV0 = { derives = [ schemars::JsonSchema ]},
-        Chipset = { derives = [ schemars::JsonSchema ]},
-        CrucibleStorageBackend = { derives = [ schemars::JsonSchema ]},
-        Cpuid = { derives = [ schemars::JsonSchema ]},
-        CpuidEntry = { derives = [ schemars::JsonSchema, PartialEq, Eq, Copy ]},
-        CpuidVendor = { derives = [ schemars::JsonSchema ]},
-        DlpiNetworkBackend = { derives = [ schemars::JsonSchema ]},
-        FileStorageBackend = { derives = [ schemars::JsonSchema ]},
-        GuestHypervisorInterface = { derives = [ schemars::JsonSchema ]},
-        I440Fx = { derives = [ schemars::JsonSchema ]},
-        HyperVFeatureFlag = { derives = [ schemars::JsonSchema ]},
-        MigrationFailureInjector = { derives = [ schemars::JsonSchema ]},
-        NvmeDisk = { derives = [ schemars::JsonSchema ]},
+        BootSettings = { derives = [ Default] },
+        CpuidEntry = { derives = [ PartialEq, Eq, Copy ]},
```

Comment:
obnoxious whitespace nits:
```suggestion
        BootSettings = { derives = [ Default ] },
        CpuidEntry = { derives = [ PartialEq, Eq, Copy ] },
```
(rustfmt presumably refuses to touch the macro...)